src/librustdoc/html/render/print_item.rs

```diff
@@ -459,6 +459,12 @@ fn item_module(w: &mut Buffer, cx: &mut Context<'_>, item: &clean::Item, items:
     }
 }
 
+/// Provides a deprecation message based on whether it is already deprecated,
+/// or whether it will be deprecated in the future
+pub(crate) fn deprecation_message(depr: &rustc_attr::Deprecation) -> &'static str {
+    if stability::deprecation_in_effect(depr) { "Deprecated" } else { "Deprecation planned" }
+}
+
```

Comment:
Should this stay here or move to a different module? Should we send a smaller value, such as an integer, in the json and decide this message when rendering search client-side instead?

Reply:
Another option is to store a separate unique list of all deprecation messages encountered, and store an index into that list per-item. Not sure if any of that is worth it, given the ratio of deprecated to non-deprecated items.